src/main/kotlin/org/rust/ide/actions/RsShowControlFlowGraphAction.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.application.PathManager
+import com.intellij.openapi.ui.Messages
+import com.intellij.psi.util.parentOfType
+import com.intellij.util.io.createDirectories
+import com.intellij.util.io.delete
+import org.rust.lang.core.cfg.ControlFlowGraph
+import org.rust.lang.core.psi.RsBlock
+import org.rust.lang.core.psi.RsFunction
+import java.awt.Desktop
+import java.io.File
+import java.io.IOException
+import java.nio.file.Paths
+
+
+class RsShowControlFlowGraphAction : AnAction() {
+    private val CONTROL_FLOW_GRAPH = "control_flow_graph"
+    private val tempDir = Paths.get(PathManager.getTempPath()).resolve(CONTROL_FLOW_GRAPH)
+
+    init {
+        tempDir.delete()
+    }
+
+    override fun actionPerformed(event: AnActionEvent) {
+        val body = getBody(event.dataContext) ?: return
+
+        val cfg = ControlFlowGraph.buildFor(body)
+        val dotDescription = cfg.createDotDescription()
+
+        val dotFileName = "cfg${cfg.hashCode()}.dot"
+        val pngFileName = "cfg${cfg.hashCode()}.png"
+
+        val dotFile = createTempFileWithContent(dotFileName, dotDescription)
+        val pngFile = tempDir.resolve(pngFileName).toFile()
+
+        try {
+            val dotFilePath = tempDir.resolve(dotFileName)
+            val pngFilePath = tempDir.resolve(pngFileName)
+            Runtime.getRuntime().exec("dot -Tpng $dotFilePath -o $pngFilePath").waitFor()
+        } catch (e: IOException) {
+            Messages.showInfoMessage(e.message, "Cannot show graph")
+        }
+
+        if (pngFile.exists()) {
+            Desktop.getDesktop().open(pngFile)
+        }
+
+        dotFile.deleteOnExit()
+        pngFile.deleteOnExit()
```

Comment:
Related to comment about tmp directory - I'm not sure that it's a good idea to delete these files because they can be useful outside of IDE in some cases

Reply:
I think it's not so useful because you can just perform the action again.